oximeter/collector/src/lib.rs

```diff
@@ -454,9 +453,39 @@ impl OximeterAgent {
                 CLICKHOUSE_PORT,
             )
         };
+
+        // Determine the version of the database.
+        //
+        // There are three cases
+        //
+        // - The database exists and is at the expected version. Continue in
+        // this case.
+        //
+        // - The database exists and is at a lower-than-expected version. We
+        // fail back to the caller here, which will retry indefinitely until the
+        // DB has been updated.
```

Comment:
Nice, so until we have support for "online upgrade", this prevents Clickhouse from ever running with an old schema, right?

Reply:
That's right. Oximeter will spin on startup until the DB version matches its compiled version.